scripts/docs_config/glossary.rst

```diff
@@ -0,0 +1,100 @@
+Glossary
+==========================================================
+
+Homogeneous Memory
+  A set of memory used in a system that is composed of a single memory type of 
+  memory technology, managed by a singular driver using a uniform approach.
+
+Heterogeneous Memory
+  A set of memory used in a system that is composed of multiple types of memory 
+  technologies, each requiring distinct handling approaches often managed by 
+  separate drivers.
+
+Memory Tiering
+  An organization of different types of memory storage within a system, each 
+  having distinct characteristics, performance, and cost attributes. These 
+  memory tiers are typically orrganized in a hierarchy, with faster, more 
```

Comment:
@lukaszstolarczuk @PatKamin yes I changed this only to check if SpellCheck catch this. It would be great if we enable SpellCheck also for *.rst and *.md files